include/uv.h

```diff
@@ -1664,6 +1677,30 @@ union uv_any_req {
 };
 #undef XX
 
+#if defined(DISCORD_ENABLE_NETMAP)
+typedef struct nm_desc nm_desc_t;
+typedef struct uv_netmap_s uv_netmap_t;
+#define NM_NUM_UDP_SOCKETS (1 << 16)
+
+struct uv_netmap_s {
+  UV_HANDLE_FIELDS
+  nm_desc_t* intf;
+  uv__io_t io_watcher;
+  uv_udp_t* sockets[NM_NUM_UDP_SOCKETS];
+  unsigned char socket_tos[NM_NUM_UDP_SOCKETS];
+  void* write_queue[2];
+  void* write_completed_queue[2];
+  unsigned char src_mac[6];
+  unsigned char dst_mac[6];
+  unsigned char src_ip[4];
```

Comment:
Unconcerned with ipv6? Should we need to assert that at init time?

Reply:
Yeah I haven't bothered at all with ipv6 at this point. Good point, I'll add an assert somewhere. If/when we want to tackle that, we'll need to also add new packet marshal/unmarshal for it.